proposals/0000-synthesized-comparable-for-enumerations.md

```diff
@@ -0,0 +1,164 @@
+{
+    case low
+    case medium
+    case high
+
+    private static
+    func minimum(_ lhs:Self, _ rhs:Self) -> Self
+    {
+        switch (lhs, rhs)
+        {
+        case (.low,    _), (_, .low   ):
+            return .low
+        case (.medium, _), (_, .medium):
+            return .medium
+        case (.high,   _), (_, .high  ):
+            return .high
+        }
+    }
+
+    static func < (lhs:Self, rhs:Self) -> Bool
+    {
+        return (lhs != rhs) && (lhs == Self.minimum(lhs, rhs))
+    }
+}
+```
+
+As the second workaround is non-obvious to many, users also often attempt to implement “private” integer values for enumeration cases by manually numbering them. Needless to say, this approach scales very poorly, and incurs a high code maintenance cost as simple tasks like adding a new enumeration case require manually re-numbering all the other cases. Workarounds for the workaround, such as numbering by tens (to “make room” for future cases) or using `Double` as the key type (to allow [numbering “on halves”](https://youtu.be/KWcxgrg4eQI?t=113)) reflect poorly on the language.
+
+```swift
+enum Membership:Comparable
+{
+    case premium
+    case preferred
+    case general
+
+    private
+    var comparisonValue:Int
+    {
+        switch self
+        {
+        case .premium:
+            return 0
+        case .preferred:
+            return 1
+        case .general:
+            return 2
+        }
+    }
+
+    static
+    func < (lhs:Self, rhs:Self) -> Bool
+    {
+        return lhs.comparisonValue < rhs.comparisonValue
+    }
+}
+```
+
+## proposed solution
+
+Enumeration types which opt-in to a synthesized `Comparable` conformance would compare according to case declaration order, with later cases comparing greater than earlier cases. Only pure `enum` types, without raw or associated values, would be eligible for synthesized conformances.
+
+While basing behavior off of declaration order is unusual for Swift, as we generally hew to the “all fields are reorderable by the compiler” principle, it is not a foreign concept to `enums`. For example, reordering cases in a numeric-backed raw `enum` already changes its runtime behavior, since the case declaration order is taken to be meaningful in that context. I also believe that `enum` cases and `struct`/`class` fields are sufficiently distinct concepts that making enumeration case order meaningful would not make the language incoherent.
+
+Later cases will compare greater than earlier cases, as Swift generally views sort orders to be “ascending” by default. It also harmonizes with the traditional C/C++ paradigm where a sequence of enumeration cases is merely a sequence of incremented integer values.
+
+## detailed design
+
+Synthesized `Comparable` conformances will work exactly the same as synthesized `Equatable`, `Hashable`, and `Codable` conformances today. A conformance will not be synthesized if a type already provides an explicit `<` implementation.
```

Comment:
Given those synthetic implementations also cover enums with associated values, it seems a bit too broad to say it will work the same.

Reply:
i meant for "eligible" type implementations,, clarified